### PR TITLE
feat: Deprecation of RemoteTargetRef in v1beta2

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**

Changes proposed in this pull request:

- Deletes the functionality of Kyma.Spec.Modules[]. RemoteTargetRef

**Related issue(s)**

### CODE REVIEW

File: pkg/templatelookup/regular.go [hunk 1]
```diff
@@ -32,7 +30,9 @@ type ModuleTemplateInfo struct {
 	DesiredChannel string
 }
 
-func NewTemplateLookup(reader client.Reader, descriptorProvider *provider.CachedDescriptorProvider, syncEnabled bool) *TemplateLookup {
+func NewTemplateLookup(reader client.Reader, descriptorProvider *provider.CachedDescriptorProvider,
+	syncEnabled bool,
```
Comment (lindnerby) [inline pkg/templatelookup/regular.go:34]: `syncEnabled` can now also be removed from `TemplateLookup` 👍

File: tests/integration/controller/controlplane/kyma_remote_sync_test.go [hunk 2]
```diff
@@ -370,16 +305,16 @@ var _ = Describe("Kyma sync default module list into Remote Cluster", Ordered, f
 
 var _ = Describe("CRDs sync to SKR and annotations updated in KCP kyma", Ordered, func() {
 	kyma := NewTestKyma("kyma-test-crd-update")
-	moduleInKcp := NewTestModule("in-kcp", v1beta2.DefaultChannel)
-	kyma.Spec.Modules = []v1beta2.Module{moduleInKcp}
-
-	remoteKyma := &v1beta2.Kyma{}
+	moduleInKCP := NewTestModule("in-kcp", v1beta2.DefaultChannel)
```
Comment (lindnerby) [inline tests/integration/controller/controlplane/kyma_remote_sync_test.go:308]: Can you reiterate over this test case please.
For example:
`moduleInKCP` gets created with name and channel, but is nowhere used except its name and channel in next line, where even the list is initialized with a new `v1beta2.Module`.

**Review by lindnerby — CHANGES_REQUESTED** [review]

File: pkg/templatelookup/regular.go (hunk 1 above)
Comment (jeremyharisch) [reply pkg/templatelookup/regular.go:34]: Nice, catch! Thanks

File: tests/integration/controller/controlplane/kyma_remote_sync_test.go (hunk 2 above)
Comment (jeremyharisch) [reply tests/integration/controller/controlplane/kyma_remote_sync_test.go:308]: It is used, as you already emtnioned in the next line. We use the `NewTestModule` function to generate a unique name. It then gets inserted into the Kyma Spec and a few lines after with `registerControlPlaneLifecycleForKyma(kyma)` a template gets created for it and deployed on the cluster. 

Indeed, we had a duplication of creating the template twice, but I now removed it

**Review by lindnerby — APPROVED** [review]

File: tests/integration/controller/controlplane/kyma_remote_sync_test.go (hunk 2 above)
Comment (lindnerby) [reply tests/integration/controller/controlplane/kyma_remote_sync_test.go:308]: Yeah but we already pass name and channel and never use the Module itself. The unique name generation is not having any effect. But nevermind. :)

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**